src/Symfony/Component/HttpFoundation/InputBag.php

```diff
@@ -73,6 +73,35 @@ public function set(string $key, mixed $value)
         $this->parameters[$key] = $value;
     }
 
+    /**
+     * Returns the parameter value converted to integer.
+     *
+     * @param bool $failOnInvalid Throw an exception when the value is not a valid int
+     */
+    public function getInt(string $key, int $default = 0, /* bool $failOnInvalid = false */): int
+    {
+        $failOnInvalid = \func_num_args() >= 3 ? func_get_arg(2) : false;
+
+        if (!$failOnInvalid) {
+            trigger_deprecation('symfony/http-foundation', '6.3', 'Non-numeric parameters will throw an exception in Symfony 7.0. Pass true as 3rd argument to "%s()" to opt-in for this behavior.', __METHOD__);
```

Comment:
WDYT of this?
`Pass true as 3rd argument to "%s()" to return a 400 when a non-numeric value is passed to the request for parameter "%s".`
(same below)